common4j/src/main/com/microsoft/identity/common/java/util/HashMapExtensions.java

```diff
@@ -43,10 +43,10 @@
         final HashMap<String, String> response = new HashMap<>();
         if (webResponse != null && !StringUtil.isNullOrEmpty(webResponse.getBody())) {
             final JSONObject jsonObject = new JSONObject(webResponse.getBody());
-            final Iterator<?> i = jsonObject.keys();
-            while (i.hasNext()) {
-                String key = (String) i.next();
-                response.put(key, jsonObject.getString(key));
+            final Iterator<String> keyIterator = jsonObject.keys();
+            while (keyIterator.hasNext()) {
+                String key = keyIterator.next();
+                response.put(key, jsonObject.get(key).toString());
```

Comment:
It could have been a json lib change that brought this on.  Did the server schema change to become error_codes:list instead of error_code:string?  Can we take on schema validation here?

Reply:
What call is being made that is interpreted in this way?

Reply:
Getting an AT in exchange for a PRT

Reply:
Our own tests with mocked responses indicate that this was always an array on the server: https://github.com/AzureAD/ad-accounts-for-android/blob/dev/AADAuthenticator/src/sharedTestClasses/com/microsoft/broker/sharedtestclasses/TestUtils.java#L397